test/e2e/service_options_test.go

```diff
@@ -167,6 +171,23 @@ func (test *e2eTest) validateServiceAnnotations(t *testing.T, r *KnRunResultColl
 	}
 }
 
+func (test *e2eTest) validateLabels(t *testing.T, r *KnRunResultCollector, serviceName string, serviceLabels, revisionLabels map[string]string) {
+	out := test.kn.Run("service", "describe", serviceName, "-ojson")
+	data := json.NewDecoder(strings.NewReader(out.Stdout))
+	var service servingv1.Service
+	err := data.Decode(&service)
+	assert.NilError(t, err)
+
+	gotRevisionLabels := service.Spec.Template.ObjectMeta.GetLabels()
+	for k, v := range revisionLabels {
+		assert.Equal(t, gotRevisionLabels[k], v)
+	}
+	gotServiceLabels := service.ObjectMeta.GetLabels()
+	for k, v := range serviceLabels {
+		assert.Equal(t, gotServiceLabels[k], v)
+	}
```

Comment:
For a full check, you would also need to check that every label in `gotServiceLabels` has been verified (i.e. that the labels returned by the service are really all stored in `gotServiceLabels`).

Imagine that `GetLabels` returns an empty map (so no label has been created, which would be a bug). But your validation would still succeed.

Actually you should _remove_ the entry in `gotServiceLabels` after each _assert_ and then check after the loop that `gotServiceLabels` is indeed empty.

Reply:
Same btw for the loop above, too.

Reply:
> Imagine that GetLabels returns an empty map (so no label has been created, which would be a bug). But your validation would still succeed.

If GetLabels returns an empty map and `serviceLabels` has any label to be validated then validation logic will fail.

Reply:
Sorry, you are right with `GetLabel` returning an empty map would lead to an error. But would happen if `GetLabel` returns _more_ labels than expected ? Is this supposed to be an error is this something that is also expected ? 

Of course this is a question for what to check. If you have full control over the service, I would expect that there are no extra labels being added to revisions and annotations (so would still suggest to check that all of the labels you get with `GetLabel` are really expected). In other words, I would check that both maps are equal not that the expected map is subset of the actual map.

Reply:
> But would happen if GetLabel returns more labels than expected ?

This is not suppose to be an error IMO. Function responsibility is to verify expectedLabels are present in received service labels, not to check both are an exact match. 

>  If you have full control over the service, I would expect that there are no extra labels being added to revisions and annotations

Thanks for explaining the pretext here. I see your argument now about checking for exact match. 👍

Reply:
I have updated the PR. Please review it at your convenience